tests/bluemira/structural/test_plotting.py::TestPlotting::test_everything_looks_good fails on develop

## Describe the bug

tests/bluemira/structural/test_plotting.py::TestPlotting::test_everything_looks_good is a longrun test and is currently failing on develop for me locally

## Steps to reproduce

Steps to reproduce the behaviour:

`pytest --longrun tests/bluemira/structural/test_plotting.py::TestPlotting::test_everything_looks_good fails`

## Expected behaviour

Test passes

## Screenshots / Tracebacks

```
_________________________________________________________________________________________________ TestPlotting.test_everything_looks_good __________________________________________________________________________________________________

self = <tests.bluemira.structural.test_plotting.TestPlotting object at 0x7fcd3f766190>

    def test_everything_looks_good(self):
    
        fem = FiniteElementModel()
    
        geometry = Geometry()
        rect_beam = RectangularBeam(0.5, 0.5)
        nodes = [[0, 0, 0], [1, 0, 0], [1, 0, 1], [0, 0, 1], [0, 0, 2], [1, 0, 2]]
        for node in nodes:
            geometry.add_node(*node)
        for i in range(5):
            geometry.add_element(i, i + 1, rect_beam, SS316)
        geometry.add_element(0, 3, rect_beam, SS316)
        geometry.add_element(2, 5, rect_beam, SS316)
    
        l_loop = Loop([2.5, 1.5, 1.5], [0, 0, 0], [0, 0, 2])
        u_loop = Loop([0, 0, 1, 1], [0, 0, 0, 0], [2, 0, 0, 2])
        u_loop.translate([3, 0, 0])
        e_loop = Loop(x=[1, 0, 0, 1, 0, 0, 1], z=[2, 2, 1.005, 1.005, 1, 0, 0])
        e_loop.translate([4.5, 0, 0])
    
        p_loop = Loop(x=[0, 0, 0, 1, 1, 0.01], z=[0, 1, 2, 2, 1, 1])
        p_loop.translate([6, 0, 0])
        r_loop = Loop(x=[0, 0, 0, 1, 1, 0.01, 1], z=[0, 1, 2, 2, 1, 1, 0])
        r_loop.translate([7.5, 0, 0])
    
        i_loop = Loop(x=[0, 0.49, 0.49, 0, 1, 0.51, 0.51, 1], z=[0, 0, 2, 2, 2, 2, 0, 0])
        i_loop.translate([9, 0, 0])
        n_loop = Loop(x=[0, 0, 1, 1], z=[0, 2, 0, 2])
        n_loop.translate([10.5, 0, 0])
        t_loop = Loop(x=[0, 0.49, 0.5, 0.51, 1], z=[2, 2, 0, 2, 2])
        t_loop.translate([12, 0, 0])
    
        for letter in [l_loop, u_loop, e_loop, p_loop, r_loop, i_loop, n_loop, t_loop]:
            letter.rotate(30, p1=[0, 0, 0], p2=[0, 0, 1])
>           geometry.add_coordinates(letter, rect_beam, SS316)

tests/bluemira/structural/test_plotting.py:69: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <bluemira.structural.geometry.Geometry object at 0x7fcd3f700070>, coordinates = <bluemira.geometry._deprecated_loop.Loop object at 0x7fcd3f64f820>
cross_section = <bluemira.structural.crosssection.RectangularBeam object at 0x7fcd3eb21eb0>
material = StructuralMaterial(E=200000000000.0, nu=0.33, rho=8910, alpha=1.8e-05, sigma_y=360000000.0, G=300751879699.2481, description='Typical stainless steel properties at room temperature')

    def add_coordinates(self, coordinates, cross_section, material=None):
        """
        Adds a Loop object to the Geometry
    
        Parameters
        ----------
        coordinates: Coordinates
            The coordinates to transform into connected Nodes and Elements
        cross_section: CrossSection
            The cross section of all the Elements in the Coordinates
        material: Material
            The material of all the Elements in the Coordinates
        """
        n_start = self.add_node(*coordinates[0])  # Add first Node
    
        n1 = n_start
>       for point in coordinates.points[1:]:
E       AttributeError: 'Loop' object has no attribute 'points'

bluemira/structural/geometry.py:330: AttributeError
```

## Environment (please complete the following information)

- OS: Ubuntu 21.10
- Version [e.g. 0.0.x] :slightly_smiling_face: 
- Branch (if developing): develop

## Additional context

Add any other context about the problem here.

--- a/bluemira/structural/geometry.py
+++ b/bluemira/structural/geometry.py
@@ -324,7 +324,7 @@ class Geometry:
         material: Material
             The material of all the Elements in the Coordinates
         """
-        n_start = self.add_node(*coordinates[0])  # Add first Node
+        n_start = self.add_node(*coordinates.points[0])  # Add first Node
 
         n1 = n_start
         for point in coordinates.points[1:]:

--- a/tests/structural/test_geometry.py
+++ b/tests/structural/test_geometry.py
@@ -24,6 +24,7 @@ import pytest
 from matplotlib import pyplot as plt
 
 import tests
+from bluemira.geometry.coordinates import Coordinates
 from bluemira.structural.crosssection import IBeam
 from bluemira.structural.geometry import Geometry
 from bluemira.structural.material import SS316
@@ -237,3 +238,11 @@ class TestMove:
         g.move_node(5, dx=0.5)
         assert g.n_nodes == 5
         assert g.n_elements == 6
+
+    def test_add_coordinates(self):
+        x_section = IBeam(0.1, 0.1, 0.025, 0.05)
+        g = Geometry()
+        c = Coordinates({"x": [0, 1, 2, 3], "z": [0, -1, 1, -1]})
+        g.add_coordinates(c, x_section, SS316)
+        assert g.n_nodes == 4
+        assert g.n_elements == 3

--- a/tests/structural/test_plotting.py
+++ b/tests/structural/test_plotting.py
@@ -60,7 +60,7 @@ class TestPlotting:
         r_loop.translate([7.5, 0, 0])
 
         i_loop = Coordinates(
-            x=[0, 0.49, 0.49, 0, 1, 0.51, 0.51, 1], z=[0, 0, 2, 2, 2, 2, 0, 0]
+            {"x": [0, 0.49, 0.49, 0, 1, 0.51, 0.51, 1], "z": [0, 0, 2, 2, 2, 2, 0, 0]}
         )
         i_loop.translate([9, 0, 0])
         n_loop = Coordinates({"x": [0, 0, 1, 1], "z": [0, 2, 0, 2]})
@@ -69,7 +69,7 @@ class TestPlotting:
         t_loop.translate([12, 0, 0])
 
         for letter in [l_loop, u_loop, e_loop, p_loop, r_loop, i_loop, n_loop, t_loop]:
-            letter.rotate(30, p1=[0, 0, 0], p2=[0, 0, 1])
+            letter.rotate(base=(0, 0, 1), direction=(0, 0, 1), degree=30)
             geometry.add_coordinates(letter, rect_beam, SS316)
 
         fem.set_geometry(geometry)

--- a/tests/structural/test_plotting.py
+++ b/tests/structural/test_plotting.py
@@ -46,24 +46,26 @@ class TestPlotting:
         geometry.add_element(0, 3, rect_beam, SS316)
         geometry.add_element(2, 5, rect_beam, SS316)
 
-        l_loop = Coordinates([2.5, 1.5, 1.5], [0, 0, 0], [0, 0, 2])
-        u_loop = Coordinates([0, 0, 1, 1], [0, 0, 0, 0], [2, 0, 0, 2])
+        l_loop = Coordinates([[2.5, 1.5, 1.5], [0, 0, 0], [0, 0, 2]])
+        u_loop = Coordinates([[0, 0, 1, 1], [0, 0, 0, 0], [2, 0, 0, 2]])
         u_loop.translate([3, 0, 0])
-        e_loop = Coordinates(x=[1, 0, 0, 1, 0, 0, 1], z=[2, 2, 1.005, 1.005, 1, 0, 0])
+        e_loop = Coordinates(
+            {"x": [1, 0, 0, 1, 0, 0, 1], "z": [2, 2, 1.005, 1.005, 1, 0, 0]}
+        )
         e_loop.translate([4.5, 0, 0])
 
-        p_loop = Coordinates(x=[0, 0, 0, 1, 1, 0.01], z=[0, 1, 2, 2, 1, 1])
+        p_loop = Coordinates({"x": [0, 0, 0, 1, 1, 0.01], "z": [0, 1, 2, 2, 1, 1]})
         p_loop.translate([6, 0, 0])
-        r_loop = Coordinates(x=[0, 0, 0, 1, 1, 0.01, 1], z=[0, 1, 2, 2, 1, 1, 0])
+        r_loop = Coordinates({"x": [0, 0, 0, 1, 1, 0.01, 1], "z": [0, 1, 2, 2, 1, 1, 0]})
         r_loop.translate([7.5, 0, 0])
 
         i_loop = Coordinates(
             x=[0, 0.49, 0.49, 0, 1, 0.51, 0.51, 1], z=[0, 0, 2, 2, 2, 2, 0, 0]
         )
         i_loop.translate([9, 0, 0])
-        n_loop = Coordinates(x=[0, 0, 1, 1], z=[0, 2, 0, 2])
+        n_loop = Coordinates({"x": [0, 0, 1, 1], "z": [0, 2, 0, 2]})
         n_loop.translate([10.5, 0, 0])
-        t_loop = Coordinates(x=[0, 0.49, 0.5, 0.51, 1], z=[2, 2, 0, 2, 2])
+        t_loop = Coordinates({"x": [0, 0.49, 0.5, 0.51, 1], "z": [2, 2, 0, 2, 2]})
         t_loop.translate([12, 0, 0])
 
         for letter in [l_loop, u_loop, e_loop, p_loop, r_loop, i_loop, n_loop, t_loop]:

--- a/tests/structural/test_plotting.py
+++ b/tests/structural/test_plotting.py
@@ -19,8 +19,9 @@
 # You should have received a copy of the GNU Lesser General Public
 # License along with bluemira; if not, see <https://www.gnu.org/licenses/>.
 
-import pytest
+import matplotlib.pyplot as plt
 
+import tests
 from bluemira.geometry.coordinates import Coordinates
 from bluemira.structural.crosssection import RectangularBeam
 from bluemira.structural.geometry import Geometry
@@ -30,9 +31,8 @@ from bluemira.structural.model import FiniteElementModel
 from bluemira.structural.plotting import GeometryPlotter
 
 
-@pytest.mark.longrun
 class TestPlotting:
-    def test_everything_looks_good(self):
+    def test_no_errors(self):
 
         fem = FiniteElementModel()
 
@@ -85,4 +85,6 @@ class TestPlotting:
 
         fem.apply_load_case(load_case)
 
-        GeometryPlotter(geometry)
+        if tests.PLOTTING:
+            GeometryPlotter(geometry)
+            plt.show()

--- a/tests/structural/test_plotting.py
+++ b/tests/structural/test_plotting.py
@@ -21,7 +21,7 @@
 
 import pytest
 
-from bluemira.geometry._deprecated_loop import Loop
+from bluemira.geometry.coordinates import Coordinates
 from bluemira.structural.crosssection import RectangularBeam
 from bluemira.structural.geometry import Geometry
 from bluemira.structural.loads import LoadCase
@@ -46,22 +46,24 @@ class TestPlotting:
         geometry.add_element(0, 3, rect_beam, SS316)
         geometry.add_element(2, 5, rect_beam, SS316)
 
-        l_loop = Loop([2.5, 1.5, 1.5], [0, 0, 0], [0, 0, 2])
-        u_loop = Loop([0, 0, 1, 1], [0, 0, 0, 0], [2, 0, 0, 2])
+        l_loop = Coordinates([2.5, 1.5, 1.5], [0, 0, 0], [0, 0, 2])
+        u_loop = Coordinates([0, 0, 1, 1], [0, 0, 0, 0], [2, 0, 0, 2])
         u_loop.translate([3, 0, 0])
-        e_loop = Loop(x=[1, 0, 0, 1, 0, 0, 1], z=[2, 2, 1.005, 1.005, 1, 0, 0])
+        e_loop = Coordinates(x=[1, 0, 0, 1, 0, 0, 1], z=[2, 2, 1.005, 1.005, 1, 0, 0])
         e_loop.translate([4.5, 0, 0])
 
-        p_loop = Loop(x=[0, 0, 0, 1, 1, 0.01], z=[0, 1, 2, 2, 1, 1])
+        p_loop = Coordinates(x=[0, 0, 0, 1, 1, 0.01], z=[0, 1, 2, 2, 1, 1])
         p_loop.translate([6, 0, 0])
-        r_loop = Loop(x=[0, 0, 0, 1, 1, 0.01, 1], z=[0, 1, 2, 2, 1, 1, 0])
+        r_loop = Coordinates(x=[0, 0, 0, 1, 1, 0.01, 1], z=[0, 1, 2, 2, 1, 1, 0])
         r_loop.translate([7.5, 0, 0])
 
-        i_loop = Loop(x=[0, 0.49, 0.49, 0, 1, 0.51, 0.51, 1], z=[0, 0, 2, 2, 2, 2, 0, 0])
+        i_loop = Coordinates(
+            x=[0, 0.49, 0.49, 0, 1, 0.51, 0.51, 1], z=[0, 0, 2, 2, 2, 2, 0, 0]
+        )
         i_loop.translate([9, 0, 0])
-        n_loop = Loop(x=[0, 0, 1, 1], z=[0, 2, 0, 2])
+        n_loop = Coordinates(x=[0, 0, 1, 1], z=[0, 2, 0, 2])
         n_loop.translate([10.5, 0, 0])
-        t_loop = Loop(x=[0, 0.49, 0.5, 0.51, 1], z=[2, 2, 0, 2, 2])
+        t_loop = Coordinates(x=[0, 0.49, 0.5, 0.51, 1], z=[2, 2, 0, 2, 2])
         t_loop.translate([12, 0, 0])
 
         for letter in [l_loop, u_loop, e_loop, p_loop, r_loop, i_loop, n_loop, t_loop]:

--- a/tests/structural/test_plotting.py
+++ b/tests/structural/test_plotting.py
@@ -48,25 +48,25 @@ class TestPlotting:
 
         l_loop = Coordinates([[2.5, 1.5, 1.5], [0, 0, 0], [0, 0, 2]])
         u_loop = Coordinates([[0, 0, 1, 1], [0, 0, 0, 0], [2, 0, 0, 2]])
-        u_loop.translate([3, 0, 0])
+        u_loop.translate((3, 0, 0))
         e_loop = Coordinates(
             {"x": [1, 0, 0, 1, 0, 0, 1], "z": [2, 2, 1.005, 1.005, 1, 0, 0]}
         )
-        e_loop.translate([4.5, 0, 0])
+        e_loop.translate((4.5, 0, 0))
 
         p_loop = Coordinates({"x": [0, 0, 0, 1, 1, 0.01], "z": [0, 1, 2, 2, 1, 1]})
-        p_loop.translate([6, 0, 0])
+        p_loop.translate((6, 0, 0))
         r_loop = Coordinates({"x": [0, 0, 0, 1, 1, 0.01, 1], "z": [0, 1, 2, 2, 1, 1, 0]})
-        r_loop.translate([7.5, 0, 0])
+        r_loop.translate((7.5, 0, 0))
 
         i_loop = Coordinates(
             {"x": [0, 0.49, 0.49, 0, 1, 0.51, 0.51, 1], "z": [0, 0, 2, 2, 2, 2, 0, 0]}
         )
-        i_loop.translate([9, 0, 0])
+        i_loop.translate((9, 0, 0))
         n_loop = Coordinates({"x": [0, 0, 1, 1], "z": [0, 2, 0, 2]})
-        n_loop.translate([10.5, 0, 0])
+        n_loop.translate((10.5, 0, 0))
         t_loop = Coordinates({"x": [0, 0.49, 0.5, 0.51, 1], "z": [2, 2, 0, 2, 2]})
-        t_loop.translate([12, 0, 0])
+        t_loop.translate((12, 0, 0))
 
         for letter in [l_loop, u_loop, e_loop, p_loop, r_loop, i_loop, n_loop, t_loop]:
             letter.rotate(base=(0, 0, 1), direction=(0, 0, 1), degree=30)

